chapter/background.md

```diff
@@ -0,0 +1,607 @@
+Applications of configuration languages are ubiquitous especially in the vicinity of software development. While XML and JSON are often used by package managers [@npm, @maven, @composer], YAML is a popular choice for complex configurations such as CI/CD pipelines [@travis, @ghaction, @gitlab-runner] or machine configurations in software defined networks such as Kubernetes and docker compose.
+
+Such static formats are used due to some significant advantages compared to other formats.
+Most strikingly, the textual representation allows inspection of a configuration without the need of a separate tool but a text editor and be version controlled using VCS software like Git.
+For software configuration this is well understood as being preferable over databases or other binary formats. Linux service configurations (files in `/etc`) and MacOS `*.plist` files which can be serialized as XML or a JSON-like format, especially exemplify that claim.
+
+Yet, despite these formats being simple to parse and widely supported [@json], their static nature rules out any dynamic content such as generated fields, functions and the possibility to factorize and reuse.
+Moreover, content validation has to be developed separately, which led to the design of complementary schema specification languages like json-schema [@json-schema] or XSD [@xsd].
+
+These qualities require an evaluated language.
+In fact, some applications make heavy use of config files written in the native programming language which gives them access to language features and existing analysis tools.
+Examples include JavaScript frameworks such as webpack [@webpack] or Vue [@vue] and python package management using `setuptools`[@setuptools].
+
+Despite this, not all languages serve as a configuration language, e.g. compiled languages and some domains require language agnostic formats.
+For particularly complex products, both language independence and advanced features are desirable.
+Alternatively to generating configurations using high level languages, this demand is addressed by more domain specific languages.
+Dhall [@dhall], Cue [@cue] or jsonnet [@jsonnet] are such domain specific languages (DSL), that offer varying support for string interpolation, (strict) typing, functions and validation.
+
+### Infrastructure as Code
+
+A prime example for the application of configuration languages are IaaS^[Infrastructure as a Service] products.
+These solutions offer great flexibility with regard to resource provision (computing, storage, load balancing, etc.), network setup and scaling of (virtual) servers.
+Although the primary interaction with those systems is imperative, maintaining entire applications' or company's environments manually comes with obvious drawbacks.
```

Comment:
```suggestion
The shift to an increasing application of IaaS^[Infrastructure as a Service] products started the desire for declarative machine configuration in a bid to simplify the setup and reproducibility of such systems.
This configuration based setup of infrastructure is commonly summarized as infrastructure as code or IaC.
As the name suggests, IaC puts cloud configuration closer to the domain of software development [@IaC-book].

In general IaaS solutions offer great flexibility with regard to resource provision (computing, storage, load balancing, etc.), network setup and scaling of (virtual) servers.
However, since the primary interaction with those systems is imperative and based on command line or web interfaces, maintaining entire applications' or company's environments manually comes with obvious drawbacks.
```